diff --git a/include/mutils/args-finder.hpp b/include/mutils/args-finder.hpp
index fdac3be..8e4c46b 100644
--- a/include/mutils/args-finder.hpp
+++ b/include/mutils/args-finder.hpp
@@ -22,7 +22,7 @@ namespace mutils{
 
 		typedef ReturnType result_type;
 
-		template <size_t i>
+		template <std::size_t i>
 		struct arg
 		{
 			typedef typename std::tuple_element<i, std::tuple<Args...>>::type type;
